packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,364 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+
+import {
+	Button,
+	Dropdown,
+	DropdownProps,
+	Input,
+	InputOnChangeData,
+	Option,
+	makeStyles,
+	shorthands,
+	tokens,
+} from "@fluentui/react-components";
+import {
+	DataEdit,
+	EditData,
+	EditType,
+	FluidObjectValueNode,
+} from "@fluid-experimental/devtools-core";
+
+import { useMessageRelay } from "../../MessageRelayContext";
+import { TreeHeader } from "./TreeHeader";
+
+/**
+ * Input to {@link EditableView}
+ */
+export interface EditableViewProps {
+	node: FluidObjectValueNode;
+	containerKey: string;
+	label: string;
+}
+
+/**
+ * Input to EditableComponents
+ */
+interface EditableComponentProps {
+	node: FluidObjectValueNode;
+	editProp: EditState | undefined;
+	setEditProp: React.Dispatch<React.SetStateAction<EditState | undefined>>;
+	submitChange: (data: EditData) => void;
+}
+
+interface EditState {
+	type: string;
+	value?: EditData;
+}
+/**
+ * Render data that is Editable with its corresponding UI
+ *
+ * @remarks {@link MessageRelayContext} must be set in order to use this component.
+ */
+export function EditableView(props: EditableViewProps): React.ReactElement {
+	const { node, containerKey, label } = props;
+
+	const [editProp, setEditProp] = React.useState<EditState | undefined>(undefined);
+
+	const messageRelay = useMessageRelay();
+
+	/**
+	 * Using the {@link fluidObjectId}, {@link containerKey}, and data posts a message to edit the DDS
+	 * @param data - The data to edit the DDS with
+	 */
+	const submitChange = React.useCallback(
+		(data: EditData): void => {
+			const edit = {
+				fluidObjectId: node.fluidObjectId,
+				data,
+			};
+
+			messageRelay.postMessage(
+				DataEdit.createMessage({
+					containerKey,
+					edit,
+				}),
+			);
+			setEditProp(undefined);
+		},
+		[containerKey, messageRelay, node.fluidObjectId],
+	);
+
+	/**
+	 * On blur will set {@link isEditing} to false
```

Comment:
Nit: This is a bit out of date. Maybe something like
```suggestion
	 * On blur will cancel any ongoing edit
```